Content.Server/Chemistry/ReagentEffects/MakeSentient.cs

```diff
@@ -0,0 +1,60 @@
+﻿using System;
+
+//using Content.Server.AI.Components; // This was in the original code but caused an error so I'm gonna keep it as a comment just in case
+using Content.Server.Mind.Components;
+using Content.Server.NPC.Components; // Don't think I actually need this but keeping it just in case
+using Content.Shared.Emoting;
+using Content.Shared.Examine;
+using Content.Shared.Movement.Components;
+using Content.Shared.Speech;
+using Content.Shared.Damage;
+
+using Content.Shared.Chemistry.Reagent;
+using Content.Server.Ghost.Roles.Components;
+
+namespace Content.Server.Chemistry.ReagentEffects;
+
+public sealed class MakeSentient : ReagentEffect
+{
+    public override void Effect(ReagentEffectArgs args)
+    {
+        var entityManager = args.EntityManager;
+        var uid = args.SolutionEntity;
+
+        // This makes it so it doesn't affect things that are already sentient
+        if (entityManager.HasComponent<MindComponent>(uid))
+        {
+            return;
+        }
+
+        // This removes the ActiveNPCComponent, because I'm pretty sure that gets rid of the preexisting AI which we probably don't want
+        // Any glitches found in this code probably stem from this part
+        if (entityManager.HasComponent<ActiveNPCComponent>(uid))
+            entityManager.RemoveComponent<ActiveNPCComponent>(uid);
+        
+        // This inactive piece of code makes things able to speak normally, but we probably don't need it
+        //if (entityManager.HasComponent<ReplacementAccentComponent>(uid))
+        //    entityManager.RemoveComponent<ReplacementAccentComponent>(uid);
+
+        entityManager.EnsureComponent<MindComponent>(uid);
+        //entityManager.EnsureComponent<SharedPlayerInputMoverComponent>(uid); // No idea what this line was for, but it caused errors and the code works fine without them
+        //entityManager.EnsureComponent<SharedPlayerMobMoverComponent>(uid); // No idea what this line was for, but it caused errors and the code works fine without them
+        entityManager.EnsureComponent<SharedSpeechComponent>(uid);
+        entityManager.EnsureComponent<SharedEmotingComponent>(uid);
+        entityManager.EnsureComponent<ExaminerComponent>(uid);
+
+        // No idea what anything past this point does
+        if (entityManager.TryGetComponent(uid, out GhostTakeoverAvailableComponent? takeOver))
+        {
+            return;
+        }
+
+        takeOver = entityManager.AddComponent<GhostTakeoverAvailableComponent>(uid);
```

Comment:
I think you can actually delete everything between the mind check and this line because that component handles all of that stuff itself. Just make a decision on what to do with the replacement access too.

Reply:
Got rid of all the bits of code that probably weren't needed, and it does in fact work (I bothered testing it this time). I also got rid of `entityManager.RemoveComponent<ActiveNPCComponent>(uid);` because as metalgearsloth said, it will become unnecessary in #12014.